src/makeInfinityAdder.js

```diff
@@ -6,7 +6,30 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let clearCall = 0;
+  let result = 0;
+
+  const adder = (arg) => {
+    if (arg === undefined) {
+      clearCall++;
+
+      if (clearCall === 1) {
+        return result;
+      }
+
+      if (clearCall === 2) {
+        result = 0;
+
+        return result;
+      }
```

Comment:
```
if (clearCall !== 0) {
        const currentResult = result;

        result = 0;

        return currentResult;
      }
```

      You can avoid using secondary "if" construction by adding a temporary variable.

Reply:
Well thank you. And in this way, I don't even need to count the number of function runs without arguments.